### PR TITLE
MAINT: Misc. typo fixes

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell`

### CODE REVIEW

File: numpy/core/src/multiarray/ctors.c [hunk 1]
```diff
@@ -3166,7 +3166,7 @@ PyArray_Zeros(int nd, npy_intp *dims, PyArray_Descr *type, int is_f_order)
  * Empty
  *
  * accepts NULL type
- * steals referenct to type
+ * steals referenced to type
```
Comment (luzpaz) [inline numpy/core/src/multiarray/ctors.c]: should this actually `reference` instead ?

Comment (mattip) [reply numpy/core/src/multiarray/ctors.c]: steals a reference to type

File: numpy/doc/indexing.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
-"""==============
+"""
+==============
 Array indexing
 ==============
```
Comment (mattip) [inline numpy/doc/indexing.py]: Note to reviewers: this should be checked in the help, html and ipython rendering

**Comment by mattip** [comment]
Most of these are in the source code, only a few are user-facing.

File: numpy/core/src/multiarray/ctors.c (hunk 1 above)
Comment (luzpaz) [reply numpy/core/src/multiarray/ctors.c]: f3ff2d6

**Comment by charris** [comment]
close/reopen

**Comment by charris** [comment]
Apparently circleCI cannot be retriggered.

**Comment by tylerjereddy** [comment]
> Apparently circleCI cannot be retriggered.

I think if you set it up on your fork and then open a PR to main repo after it won't even show up in the CI list. Pretty sure SciPy recently starting CircleCI to do merge tests instead of branch only---not sure if that would help with what you're trying to accomplish here.

File: numpy/doc/indexing.py (hunk 2 above)
Comment (eric-wieser) [reply numpy/doc/indexing.py]: The other files in this directory are all wrapped this way too, so I think fine to skip the check.

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @luzpaz. I wonder if we could set up some commit hooks for codespell, and flake8 to prevent these from creeping in

**Comment by mattip** [comment]
convincing CircleCI to do merge tests sounds good. @tylerjereddy do you know what SciPy did?

**Comment by luzpaz** [comment]
> Thanks @luzpaz. I wonder if we could set up some commit hooks for codespell, and flake8 to prevent these from creeping in

@mattip My pleasure :smile: BTW, checkout https://github.com/codespell-project/codespell/issues/139

**Pull request merged**